Kentor.AuthServices/Saml2StatementExtension.cs

```diff
@@ -69,9 +80,18 @@ private static XElement ToXElement(Saml2AttributeStatement attributeStatement)
                 attributeElement.AddAttributeIfNotNullOrEmpty("NameFormat", attribute.NameFormat);
                 attributeElement.AddAttributeIfNotNullOrEmpty("OriginalIssuer", attribute.OriginalIssuer);
 
+                XNamespace xs = @"http://www.w3.org/2001/XMLSchema";
+                XNamespace xsi = @"http://www.w3.org/2001/XMLSchema-instance";
+
                 foreach (var value in attribute.Values)
                 {
-                    attributeElement.Add(new XElement(Saml2Namespaces.Saml2 + "AttributeValue", value));
+                    if (enforceXmlns)
+                        attributeElement.Add(new XElement(Saml2Namespaces.Saml2 + "AttributeValue", value,

```

Comment:
Please use {} on all if statements. Also rewrite this to create the new XElement first and then add the XNamespace attributes inside an if-block. Don't want the basic attributeElement.Add line duplicated as it can make bugs appear if one is changed and not the other. And having it twice means tests must be duplicated to ensure functional coverage.